initrd/etc/functions

```diff
@@ -697,8 +697,17 @@ mount_possible_boot_device() {
 		return 1
 	fi
 
-	TRACE "Try mounting $BOOT_DEV as /boot"
-	if mount -o ro "$BOOT_DEV" /boot >/dev/null 2>&1; then
+	# Get the size of BOOT_DEV in 512-byte sectors
+	sectors=$(blockdev --getsz "$BOOT_DEV")
+
+	# Check if the partition is small (less than 2MB, which is 4096 sectors)
+	if [ "$sectors" -lt 4096 ]; then
+		TRACE_FUNC
+		DEBUG "Partition $BOOT_DEV is very small, likely BIOS boot. Skipping mount."
+		return 1
+	TRACE_FUNC
+	DEBUG "Try mounting $BOOT_DEV as /boot"
+	elif mount -o ro "$BOOT_DEV" /boot >/dev/null 2>&1; then
```

Comment:
```suggestion
		TRACE_FUNC
		DEBUG "Try mounting $BOOT_DEV as /boot"
	elif mount -o ro "$BOOT_DEV" /boot >/dev/null 2>&1; then
```

@tlaurion Could you please fix the indentation here?

Reply:
@JonathonHall-Purism the identation os fine, since the TRACE_FUNC and DEBUG are placed i such indent because they are meant to show where and what will be done under that elif calling an action. The indent is aimed at showing to the reader that the next call is action, not linked to prior code block. Makes sense?

Reply:
No, it doesn't make sense :thinking: 

As written, it will **not** print "Try mounting $BOOT_DEV as /boot" when it is about to try mounting $BOOT_DEV as /boot.  It will **only** print "Try mounting $BOOT_DEV as /boot" when it is **not** about to try mounting $BOOT_DEV as /boot.

That is the opposite of what you want.

Clearly I didn't read the message well enough to connect it to the statement below, that's my fault.  Fixing the indentation reveals that the code is wrong, but actually the code needs to be reworked to be correct.

Reply:
@JonathonHall-Purism fixed under [1724584](https://github.com/linuxboot/heads/pull/1784/commits/1724584fecaaf512ef707b2a7e071b328b22c13c)